runner/ollamarunner/runner.go

```diff
@@ -130,16 +130,21 @@ func (s *Server) NewSequence(prompt string, images []llm.ImageData, params NewSe
 	// Ensure that at least 1 input can be discarded during shift
 	params.numKeep = min(params.numKeep, s.cache.numCtx-1)
 
-	if int32(len(inputs)) > s.cache.numCtx {
-		discard := int32(len(inputs)) - s.cache.numCtx
+	// Basic runner-side guard: if truncation is enabled and num_ctx <= 1, return an error
+	if params.truncate && s.cache.numCtx <= 1 {
+		return nil, fmt.Errorf("input after truncation exceeds maximum context length")
+	}
 
+	if int32(len(inputs)) > s.cache.numCtx {
 		if !params.truncate {
 			return nil, errorInputTooLong
 		}
 
+		// Embeddings and generation both return the same too-long error when truncate is enabled
+
+		discard := int32(len(inputs)) - s.cache.numCtx
```

Comment:
I don't think we have to move this line from below the `if int32(len(inputs)) > s.cache.numCtx`. Furthermore the comment here isn't super relevant to the code and can be removed